### PR TITLE
gh-113664: Improve style of Big O notation (math)

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Use :math: to render them as mathematic formulas.

* Issue: gh-113664

----
📚 Documentation preview 📚: https://cpython-previews--113724.org.readthedocs.build/

### CODE REVIEW

**Comment by hugovk** [comment]
Does `:math:` also work with PDF, EPUB and plain text builds? 

https://docs.python.org/3/download.html

File: Doc/library/select.rst
Comment (erlend-aasland) [inline Doc/library/select.rst:1]: Digression: the Big O notation in `select.rst` is ... untraditional. I've never seen `O(n)` written as `O(name of some variable)`. IIRC, all off these should be written as `O(n)`. One can of course clarify `n` in a follow-up sentence in the docs.

**Comment by serhiy-storchaka** [comment]
I cannot build EPUB due to unrelated error.

In PDF it renders naturally, except cases in select.rst where spaces between words are swallowed: $`O(highest file descriptor)`$. I tried different variants, like `\text{}` or `\mathrm{}`, but they do not work in HTML. Perhaps the best way is to simply use italic font here.

In plain text it is rendered literally, e.g. ``` O(n\log{n}) ```.

**Comment by hugovk** [comment]
This PR with EPUB (in macOS Books):

image

image

image

**Pull request closed without merging**